Apps/phciscoumbrella/ciscoumbrella_connector.py

```diff
@@ -0,0 +1,401 @@
+
+        if (r.status_code == 202):  # success, return from here, requests treats 202 as !ok
+            return (phantom.APP_SUCCESS, resp_json, status_code)
+
+        if (r.status_code != requests.codes.ok):  # pylint: disable=E1101
+            return (action_result.set_status(phantom.APP_ERROR, CISCOUMB_ERR_FROM_SERVER, status=r.status_code,
+                message=self._get_error_message(resp_json, r)), resp_json, status_code)
+
+        return (phantom.APP_SUCCESS, resp_json, status_code)
+
+    def _test_connectivity(self, param):
+
+        # Progress
+        self.save_progress(CISCOUMB_USING_BASE_URL, base_url=self._base_url)
+
+        # Connectivity
+        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, self._host)
+
+        endpoint = '/domains'
+
+        action_result = ActionResult()
+
+        self.save_progress(CISCOUMB_MSG_GET_DOMAIN_LIST_TEST)
+
+        ret_val, response, status_code = self._make_rest_call(endpoint, action_result, {'page': 1, 'limit': 1})
+
+        if (phantom.is_fail(ret_val)):
+            self.debug_print(action_result.get_message())
+            self.set_status(phantom.APP_ERROR, action_result.get_message())
+            self.append_to_message(CISCOUMB_ERR_CONNECTIVITY_TEST)
+            return self.get_status()
+
+        return self.set_status_save_progress(phantom.APP_SUCCESS, CISCOUMB_SUCC_CONNECTIVITY_TEST)
+
+    def _list_blocked_domains(self, param):
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Progress
+        self.save_progress(CISCOUMB_USING_BASE_URL, base_url=self._base_url)
+
+        # Connectivity
+        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, self._host)
+
+        # Progress
+        self.save_progress(CISCOUMB_USING_BASE_URL, base_url=self._base_url)
+
+        # Connectivity
+        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, self._host)
+
+        endpoint = '/domains'
+
+        page_index = param.get(CISCOUMB_JSON_PAGE_INDEX, CISCOUMB_DEFAULT_PAGE_INDEX)
+        domain_limit = param.get(CISCOUMB_JSON_DOMAIN_LIMIT, CISCOUMB_DEFAULT_DOMAIN_LIMIT)
+
+        ret_val, response, status_code = self._make_rest_call(endpoint, action_result, {'page': page_index, 'limit': domain_limit})
+
+        if (phantom.is_fail(ret_val)):
+            self.debug_print(action_result.get_message())
+            return self.set_status(phantom.APP_ERROR, action_result.get_message())
+
+        domain_list = response.get('data', [])
+
+        action_result.update_summary({CISCOUMB_JSON_TOTAL_DOMAINS: len(domain_list)})
+
+        for domain in domain_list:
+            action_result.add_data(domain)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _unblock_domain(self, param):
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Progress
+        self.save_progress(CISCOUMB_USING_BASE_URL, base_url=self._base_url)
```

Comment:
What is the point of these progress statements being repeated?  Same goes for 'block_domain' action?

Reply:
Hi @tonyc-phantom the repeating progress statements were unnecessary, hence, removed them. Please review the MR.